packages/lesswrong/lib/collections/tags/schema.ts

```diff
@@ -670,34 +670,54 @@ const schema: SchemaType<"Tags"> = {
     graphQLtype: '[MultiDocument!]!',
     canRead: ['guests'],
     optional: true,
-    resolver: async (tag: DbTag, args: void, context: ResolverContext) => {
+    graphqlArguments: `lensSlug: String, version: String`,
+    resolver: async (tag: DbTag, { lensSlug, version }, context: ResolverContext) => {
       const { MultiDocuments, Revisions } = context;
-      const multiDocuments = await MultiDocuments.find({ parentDocumentId: tag._id, collectionName: 'Tags', fieldName: 'description' }, { sort: { index: 1 } }).fetch();
-      const revisions = await Revisions.find({ _id: { $in: multiDocuments.map(md => md.contents_latest) } }).fetch();
+      const multiDocuments = await MultiDocuments.find(
+        { parentDocumentId: tag._id, collectionName: 'Tags', fieldName: 'description' },
+        { sort: { index: 1 } }
+      ).fetch();
 
-      return multiDocuments.map(md => ({
-        ...md,
-        contents: revisions.find(r => r._id === md.contents_latest),
+      return Promise.all(multiDocuments.map(async md => {
+        let contents;
+        if (md.slug === lensSlug && version) {
+          // Fetch the specified version of the contents
+          contents = await Revisions.findOne({ documentId: md._id, version });
+        } else {
+          // Fetch the latest contents
+          contents = await Revisions.findOne({ _id: md.contents_latest });
+        }
```

Comment:
I'd probably rewrite this into a single `Revisions.find` like...
```
if (version) {
  const versionedLensId = multiDocuments.find(md => md.slug === lensSlug)?._id;
  if (versionedLensId) {
    const nonVersionedLensRevisionIds = multiDocuments.filter(md => md._id !== versionedLensId).map(md => md.contents_latest);
    const selector = {
      $or: [
        { documentId: versionedLensId, version },
        { _id: { $in: nonVersionedLensRevisionIds } }
      ]
    };
    const revisions = await Revisions.find(selector).fetch();
    return accessFilterMultiple(...);
  }
}

const revisionIds = multiDocuments.map(md => md.contents_latest);
const revisions = await Revisions.find({ _id: { $in: revisionIds } }).fetch();
return accessFilterMultiple(...);

Reply:
Also I think you need to fix the slug checking logic to look at oldSlugs as well